source/core/slang-string-util.cpp

```diff
@@ -243,4 +243,28 @@ ComPtr<ISlangBlob> StringUtil::createStringBlob(const String& string)
     }
 }
 
+/* static */bool StringUtil::areLinesEqual(const UnownedStringSlice& a, const UnownedStringSlice& b)
+{
+    List<UnownedStringSlice> slicesA;
```

Comment:
I think this is obvious, but we could avoid the allocation of these arrays by having a `getNextLine` sort of utility like:

```
UnownedStringSlice getNextLine(UnownedStringSlice& ioBuffer);
```

That would would allow writing `areLinesEqual` without any additional allocoation, and the `calcLines` function could also be built on top of the same buildling block so that it shares matching line-splitting behavior.